vulture_os/authentication/ldap/tools.py

```diff
@@ -0,0 +1,175 @@
+
+
+def find_group(ldap_repo, group_dn, attr_list):
+    client = ldap_repo.get_client()
+    group = client.search_by_dn(group_dn, attr_list=attr_list)
+
+    dn, attrs = group[0]
+    group = {"dn": dn}
+
+    for key in AVAILABLE_GROUP_KEYS:
+        ldap_key = getattr(ldap_repo, key)
+        if ldap_key:
+            group[ldap_key] = attrs[ldap_key]
+
+    group[ldap_repo.group_member_attr] = attrs[ldap_repo.group_member_attr]
+    return group
+
+def search_users(ldap_repo, search, by_dn=False):
+    client = ldap_repo.get_client()
+    tmp_users = client.search_user(f"{search}*", attr_list=["+","*"])
+    data = []
+
+    for dn, attrs in tmp_users:
+        user_attr  = attrs.get(ldap_repo.user_attr)
+        if not by_dn:
+            data.append(user_attr[0])
+        else:
+            data.append(dn)
+
+    return data
+
+def get_users(ldap_repository, group_dn):
+    group = find_group(ldap_repository, group_dn, ['*'])
+    members  = []
+    for member_dn in group['member']:
+        members.append(find_user(ldap_repository, member_dn, ["*"]))    
+    return members
+
+
+def get_groups(ldap_repository):
+    ldap_client = ldap_repository.get_client()
+    group_base_dn = f"{ldap_repository.group_dn},{ldap_repository.base_dn}"
+    data = []
+    for group_dn in ldap_client.enumerate_groups():
+        if group_base_dn not in group_dn:
+            continue
+
+        group = find_group(ldap_repository, group_dn, ['*'])
+        data.append(group)
+    return data
+
+
+def create_group(ldap_repository, group_name, members_username):
+    group_dn = f"{ldap_repository.group_attr}={group_name},{ldap_repository.group_dn},{ldap_repository.base_dn}"
+
+    client = ldap_repository.get_client()
+
+    members = []
+    for member in members_username:
+        try:
+            user = search_users(ldap_repository, member, by_dn=True)[0]
+        except KeyError:
+            return False, f"User {member} does not exist"
+        
+        members.append(user)
+
+    attrs = {
+        ldap_repository.group_attr: [group_name],
+        "objectClass": ["groupOfNames", "top"],
+        "member": members
+    }
+    r = client.add_group(group_dn, attrs)
+    logger.info(f"Group {group_dn} created in LDAP {ldap_repository.name}")
+    return True, r
+
+
+def create_user(ldap_repository, group_dn, user_name, userPassword, attrs):
+    user_dn = f"{ldap_repository.user_attr}={user_name},{group_dn}"
+    user = {
+        "sn": [user_name],
+        "cn": [user_name],
+        ldap_repository.user_attr: [user_name],
+        "objectClass": ["inetOrgPerson", "top"],
+        "description": ["User created by Vulture"]
+    }
+
+    user.update(attrs)
+    client = ldap_repository.get_client()
+    r = client.add_user(user_dn, user, group_dn, userPassword)
+    logger.info(f"User {user_name} created in LDAP {ldap_repository.name}")
+    return r
+
+
+def update_user(ldap_repository, group_dn, user_name, attrs, userPassword):
+    old_user = None
```

Comment:
why not get the group dn from the group_name, like with `get_groups()`or `create_group()` to avoid using `create_group_dn()`?